packages/nimble-components/src/menu-button/testing/menu-button.pageobject.ts

```diff
@@ -0,0 +1,84 @@
+import {
+    keyArrowDown,
+    keyArrowUp,
+    keyEnter,
+    keyEscape,
+    keySpace
+} from '@microsoft/fast-web-utilities';
+import type { MenuButton } from '..';
+import { createEventListener } from '../../utilities/tests/component';
+
+/**
+ * Page object for `nimble-menu-button` component to provide consistent ways
+ * of query and interacting with the component during tests.
+ */
+export class MenuButtonPageObject {
+    public constructor(protected readonly menuButtonElement: MenuButton) {}
+
+    public isOpen(): boolean {
+        return this.menuButtonElement.open;
+    }
+
+    public clickMenuButton(): void {
+        // Focus the menu button before calling click() because ordinarily a mouse click
+        // would bring focus to the button, but calling click() directly does not.
+        this.menuButtonElement.focus();
+        this.menuButtonElement.toggleButton!.control.click();
+    }
+
+    public async openMenu(): Promise<void> {
+        if (this.isOpen()) {
+            return;
+        }
+
+        const toggleListener = createEventListener(
+            this.menuButtonElement,
+            'toggle'
+        );
+        this.clickMenuButton();
+        await toggleListener.promise;
+    }
+
+    public pressEnter(): void {
+        const event = new KeyboardEvent('keypress', {
+            key: keyEnter
+        } as KeyboardEventInit);
+        this.menuButtonElement.toggleButton!.control.dispatchEvent(event);
```

Comment:
It's not a huge deal, but if keeping all these individual keypress funtions, I'd create a protected `sendKeypressEventToToggleButton` helper and turn all these `press___` functions into one-liners.

Reply:
As a note, the implementations of these functions aren't all quite the same, which is why I decided to keep them separate. The enter & space keys get sent to the toggle button's _control_ and the up & down arrow keys get sent to the toggle button itself.

But, I'll take a pass at factoring out common logic to see if that makes the code more readable.

Reply:
Update: As I was looking at this, I actually remembered that there is another difference -- some events need to be `keypress` and others need to be `keydown`. Given the nuances of firing the events, I'm very happy to finally encapsulate the logic within the page object. However, I don't think there is a lot to gain by creating a helper function to create & dispatch the event.